src/file/tokio_compat_file.rs

```diff
@@ -67,6 +68,8 @@ pub struct TokioCompatFile {
     buffer_len: NonZeroUsize,
     buffer: BytesMut,
 
+    write_len: AtomicUsize,
```

Comment:
Please don't use atomic here since we will always hold a mutable reference to `TokioCompatFile` when writing:

```suggestion
    write_len: usize,
```